packages/demand-bloc-trip-create/src/AutocompleteBloc.ts

```diff
@@ -62,17 +62,14 @@ export class AutocompleteBloc implements TripCreateFieldItem {
   }
 
   private getFeatchedResults() {
-    const { minLengthToSearch, autocompleteDebounceTime } = this.options
-    const isValidLength = (text: string) => text.length > minLengthToSearch
+    const { minLengthToSearch = 2, autocompleteDebounceTime } = this.options
 
     return this.query$.pipe(
       filter(query => !query.isPrefill),
       map(query => query.value),
+      filter(query => query.length > minLengthToSearch),
       debounceTime(autocompleteDebounceTime),
-      distinctUntilChanged(),
-      switchMap(val =>
-        isValidLength(val) ? this.getAddressAutocompleteData(val) : of(defaultAutocompleteData)
-      )
+      switchMap(val => this.getAddressAutocompleteData(val))
```

Comment:
This probably would work the same:
`switchMap(this.getAddressAutocompleteData)`

Reply:
nah, in that case, I will need to bind `this`